packages/core/contracts/financial-templates/common/financial-product-libraries/CoveredCallFinancialProductLibrary.sol

```diff
@@ -0,0 +1,99 @@
+     * @param financialProduct address of the financial product.
+     * @param strikePrice the strike price for the covered call to be applied to the financial product.
+     * @dev Note: a) Only the owner (deployer) of this library can set new strike prices b) A strike price cannot be 0.
+     * c) A strike price can only be set once to prevent the deployer from changing the strike after the fact.
+     * d)  financialProduct must exposes an expirationTimestamp method.
+     */
+    function setFinancialProductStrike(address financialProduct, FixedPoint.Unsigned memory strikePrice)
+        public
+        onlyOwner
+        nonReentrant()
+    {
+        require(strikePrice.isGreaterThan(0), "Cant set 0 strike");
+        require(financialProductStrikes[financialProduct].isEqual(0), "Strike already set");
+        require(ExpiringContractInterface(financialProduct).expirationTimestamp() != 0, "Invalid EMP contract");
+        financialProductStrikes[financialProduct] = strikePrice;
+    }
+
+    /**
+     * @notice Returns the strike price associated with a given financial product address.
+     * @param financialProduct address of the financial product.
+     * @return strikePrice for the associated financial product.
+     */
+    function getStrikeForFinancialProduct(address financialProduct)
+        public
+        view
+        nonReentrantView()
+        returns (FixedPoint.Unsigned memory)
+    {
+        return financialProductStrikes[financialProduct];
+    }
+
+    /**
+     * @notice Returns a transformed price by applying the call option payout structure.
+     * @param oraclePrice price from the oracle to be transformed.
+     * @param requestTime timestamp the oraclePrice was requested at.
+     * @return transformedPrice the input oracle price with the price transformation logic applied to it.
+     */
+    function transformPrice(FixedPoint.Unsigned memory oraclePrice, uint256 requestTime)
+        public
+        view
+        override
+        nonReentrantView()
+        returns (FixedPoint.Unsigned memory)
+    {
+        FixedPoint.Unsigned memory strike = financialProductStrikes[msg.sender];
+        require(strike.isGreaterThan(0), "Caller has no strike");
+        // If price request is made before expiry, return 1. Thus we can keep the contract 100% collateralized with
+        // each token backed 1:1 by collateral currency.
+        if (requestTime < ExpiringContractInterface(msg.sender).expirationTimestamp()) {
+            return FixedPoint.fromUnscaledUint(1);
+        }
+        if (oraclePrice.isLessThanOrEqual(strike)) {
+            return FixedPoint.fromUnscaledUint(0);
+        } else {
+            // Token expires to be worth the fraction of an collateral token that's in the money.
+            // eg if ETHUSD is $500 and strike is $400, token is redeemable for 100/500 = 0.2 WETH (worth $100).
+            return (oraclePrice.sub(strike)).div(oraclePrice);
```

Comment:
Is it worth adding a comment either here that we're assuming that `oraclePrice != 0`? [The EMP does allow the `oraclePrice` to be set to 0.](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/expiring-multiparty/PricelessPositionManager.sol#L829)

Reply:
It's a great point. It definitely deserves a comment. I think it's impossible to get here if oraclePrice == 0.

If I'm thinking about this correctly, if oraclePrice == 0, then strike is always >= oraclePrice, and the if above is always true, meaning this function just returns 0. Do you agree?